internals/overlord/servstate/manager.go

```diff
@@ -568,15 +574,15 @@ func servicesToStop(m *ServiceManager) ([]string, error) {
 		return nil, err
 	}
 
-	// Filter down to only those that are running.
+	// Filter down to only those that are running or in backoff
 	m.servicesLock.Lock()
 	defer m.servicesLock.Unlock()
-	var running []string
+	var notStopped []string
 	for _, name := range stop {
 		s := m.services[name]
-		if s != nil && s.state == stateRunning {
-			running = append(running, name)
+		if s != nil && (s.state == stateRunning || s.state == stateBackoff) {
```

Comment:
With this test, it looks like it's possible to miss services which are in the process of being restarted by the reaper (`stateStarting`), or in the process of terminating (`stateTerminating` and `stateKilling`), which would then go into backoff.

Reply:
```
        m.servicesLock.Lock()
        defer m.servicesLock.Unlock()
        var notStopped []string
        for _, name := range stop {
                s := m.services[name]
                if s != nil && (s.state == stateRunning || s.state == stateBackoff) {
                        notStopped = append(notStopped, name)
                }
        }
        return notStopped, nil

```

Note the first two lines - its a servstate wide lock that ensures only one thread is changing the captured state of any service at any one time.